core/src/main/java/org/apache/calcite/rel/rules/IntersectToSemiJoinRule.java

```diff
@@ -34,9 +34,52 @@
 import java.util.List;
 
 /**
- * Planner rule that translates a {@link org.apache.calcite.rel.core.Intersect}
+ * Planner rule that translates a {@link Intersect}
  * to a series of {@link org.apache.calcite.rel.core.Join} that type is
- * {@link org.apache.calcite.rel.core.JoinRelType#SEMI}.
+ * {@link JoinRelType#SEMI}. This rule supports n-way Intersect conversion,
+ * as this rule can be repeatedly applied during query optimization to
+ * refine the plan.
+ *
+ * <h2>Example</h2>
+ *
+ <p>Original sql:
```

Comment:
Is this the most general example for this rule?
If so, please explain in the comments the constraints on the supported predicates - e.g., they all have to involve the same field.
If not, please provide a more general example where the predicates look very different

Reply:
Done!